interfaces/connection.go

```diff
@@ -0,0 +1,245 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2017 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package interfaces
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/snap"
+)
+
+// Connection represents a connection between ConnectedPlug and ConnectedSlot
+type Connection struct {
```

Comment:
is this returned somewhere atm? or should it be internal for now?

Reply:
Not yet (it's only used internally in repo in the followup PR), but it will most likely be returned when repo grows getter method(s) to get connections.

Reply:
I'd say `Connection represents connection between a particular plug and slot.`